blockchain/blockchain.go

```diff
@@ -2142,7 +2152,24 @@ func (bc *BlockChain) ApplyTransaction(chainConfig *params.ChainConfig, author *
 	receipt.Logs = statedb.GetLogs(tx.Hash())
 	receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
 
-	return receipt, gas, err
+	return receipt, gas, internalTrace, err
+}
+
+func GetInternalTxTraces(tracer vm.Tracer) (*vm.InternalTxTrace, error) {
+	var (
+		internalTxTrace *vm.InternalTxTrace
+		err             error
+	)
+	switch tracer := tracer.(type) {
+	case *vm.InternalTxTracer:
+		internalTxTrace, err = tracer.GetResult()
+		if err != nil {
+			return nil, err
+		}
+	default:
+		logger.Warn("To trace internal transactions, VM tracer type should be vm.InternalTxTracer", "actualType", reflect.TypeOf(tracer).String())
```

Comment:
@jeongkyun-oh Will you allow `vm.InternalTxTace` to become `nil`? This case returns (nil, nil)

Reply:
Hmm. Well, I think the type of tracer must be `vm.InternalTxTracer`. So, all the other cases should return an error. I modified default case to return an error instead of warn logging.

Reply:
Applied on 30b15fa5bf052fb58cc95114d647afd80c605ff0